### DISCUSSION TITLE
Breaking changes 7.17 > 7.20

### DISCUSSION THREAD

**Discussion by ghost** [dlemstra/Magick.NET#678] [Help] [answered]
I updated the nuget from 7.17 to 7.20 and there are some code breaking changes. (not expected from versioning number).

Can someone please point me how to do the following as of now (didn't find any information)

**+ Cloning a magickImage ?**  
the method "Clone()" has gone totally
`IMagickImage image = rawImage.Clone())`

**+ Setting the image background** 
` image.BackgroundColor = new MagickColor(MagickColors.Transparent);`

**+ reading using a stream and settings**
method gone as well
`image.Read(Stream content, MagickReadSettings settings);`

eg, for eps, i need to set a density in the read settings before reading the strem...
` MagickReadSettings settings = new MagickReadSettings
                {
                    Density = new Density(300, 300, DensityUnit.PixelsPerInch)
                };`

Thanks !!

**Comment by dlemstra** [comment, marked as answer]
You might be expecting semver but in the documentation (https://github.com/dlemstra/Magick.NET) you can find the version strategy that explains this.

The return type of rawImage.Clone() depends on the type you used for rawImage. I suspect you are using `IMagickImage`. The type has changed to `IMagickImage<QuantumType>` where QuantumType depends on the version of the package that you are using. Q8 = `byte`, Q16 = `ushort and Q16-HDRI = `float`. You will need to change that to `IMagickImage<QuantumType>` or use `var` where possible to avoid issues. You are experiencing the same issue in all those places that you mention.

**Comment by ghost** [comment]
Thanks Dirk.

Versionning makes sense indeed.
Dunno why we used the interface ImagickImage instead of the MagickImage class in our code, i just changed it and all good.

Cheers,

Luc
